manifests/init.pp

```diff
@@ -67,18 +67,20 @@
       default: { $ilogen = 1 }
     }
     exec{"/sbin/hponcfg -f ${settingsfile} -l ${logfile}":
+
+    if $manage_package {
+      package { 'hponcfg':
+        ensure => 'present',
+        before => Exec["hponcfg -f ${settingsfile} -l ${logfile}"],
+      }
+    }
+
       onlyif      => 'test -e /sbin/hponcfg',
       path        =>'/bin:/usr/sbin:/usr/bin',
       refreshonly => true,
       subscribe   => File[$settingsfile],
       timeout     => 0,
     }
-    
-    if ! defined(Package['hponcfg']) {
```

Comment:
Yeah, I wrapped it in a parameter instead, so if someone had it defined already somewhere else they can use `manage_package => false` in the class definition. 

Long reasoning here: http://stackoverflow.com/questions/26205727/duplicate-declaration-of-same-resource-defined-in-separate-classes/26215336#26215336